### PR TITLE
Fix Tests: Added a new step and changed Step to ErrT for err logs

### PR DESCRIPTION
**Pull request by spowelljr**
Closes: #10591 & #10607

Registered a new step for a missing instance and changed `Step` to `ErrT` when outputing error logs.

Added a new outputting func `Styled` that does the styling of `Step` without the step logic.

Replaces all non-step usages of `Step` with `Styled`.

### CODE REVIEW

**Review by sharifelgamal — APPROVED** [review]

File: pkg/minikube/out/out.go [hunk 1]
```diff
@@ -84,12 +84,25 @@ func Step(st style.Enum, format string, a ...V) {
...
 		return
 	}
 	register.RecordStep(outStyled)
+	Styled(st, format, a...)
+}
+
+// Styled writes a stylized and templated message to stdout
+func Styled(st style.Enum, format string, a ...V) {
+	if JSON {
+		return
```
Comment (priyawadhwa) [inline pkg/minikube/out/out.go:99]: I'd send this to Info or add a new "Styled" JSON type

File: cmd/minikube/cmd/config/disable.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ var addonsDisableCmd = &cobra.Command{
 		if err != nil {
 			exit.Error(reason.InternalDisable, "disable failed", err)
 		}
-		out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
+		out.Styled(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
```
Comment (priyawadhwa) [inline cmd/minikube/cmd/config/disable.go:45]: out.Step

File: cmd/minikube/cmd/delete.go [hunk 3]
```diff
@@ -202,7 +202,7 @@ func purgeMinikubeDirectory() {
 	if err := os.RemoveAll(localpath.MiniPath()); err != nil {
 		exit.Error(reason.HostPurge, "unable to delete minikube config folder", err)
 	}
-	out.Step(style.Deleted, "Successfully purged minikube directory located at - [{{.minikubeDirectory}}]", out.V{"minikubeDirectory": localpath.MiniPath()})
+	out.Styled(style.Deleted, "Successfully purged minikube directory located at - [{{.minikubeDirectory}}]", out.V{"minikubeDirectory": localpath.MiniPath()})
```
Comment (priyawadhwa) [inline cmd/minikube/cmd/delete.go:205]: out.step

File: cmd/minikube/cmd/delete.go [hunk 4]
```diff
@@ -239,7 +239,7 @@ func deleteProfile(ctx context.Context, profile *config.Profile) error {
 
 		// if driver is oci driver, delete containers and volumes
 		if driver.IsKIC(profile.Config.Driver) {
-			out.Step(style.DeletingHost, `Deleting "{{.profile_name}}" in {{.driver_name}} ...`, out.V{"profile_name": profile.Name, "driver_name": profile.Config.Driver})
+			out.Styled(style.DeletingHost, `Deleting "{{.profile_name}}" in {{.driver_name}} ...`, out.V{"profile_name": profile.Name, "driver_name": profile.Config.Driver})
```
Comment (priyawadhwa) [inline cmd/minikube/cmd/delete.go:242]: out.step

File: cmd/minikube/cmd/delete.go [hunk 5]
```diff
@@ -364,7 +364,7 @@ func profileDeletionErr(cname string, additionalInfo string) error {
 }
 
 func uninstallKubernetes(api libmachine.API, cc config.ClusterConfig, n config.Node, bsName string) error {
-	out.Step(style.Resetting, "Uninstalling Kubernetes {{.kubernetes_version}} using {{.bootstrapper_name}} ...", out.V{"kubernetes_version": cc.KubernetesConfig.KubernetesVersion, "bootstrapper_name": bsName})
+	out.Styled(style.Resetting, "Uninstalling Kubernetes {{.kubernetes_version}} using {{.bootstrapper_name}} ...", out.V{"kubernetes_version": cc.KubernetesConfig.KubernetesVersion, "bootstrapper_name": bsName})
```
Comment (priyawadhwa) [inline cmd/minikube/cmd/delete.go:367]: out.step

File: cmd/minikube/cmd/delete.go [hunk 6]
```diff
@@ -442,7 +442,7 @@ func handleMultipleDeletionErrors(errors []error) {
 func deleteProfileDirectory(profile string) {
 	machineDir := filepath.Join(localpath.MiniPath(), "machines", profile)
 	if _, err := os.Stat(machineDir); err == nil {
-		out.Step(style.DeletingHost, `Removing {{.directory}} ...`, out.V{"directory": machineDir})
+		out.Styled(style.DeletingHost, `Removing {{.directory}} ...`, out.V{"directory": machineDir})
```
Comment (priyawadhwa) [inline cmd/minikube/cmd/delete.go:445]: out.step

File: cmd/minikube/cmd/generate-docs.go [hunk 7]
```diff
@@ -47,7 +47,7 @@ var generateDocs = &cobra.Command{
 		if err := generate.Docs(RootCmd, path); err != nil {
 			exit.Error(reason.InternalGenerateDocs, "Unable to generate docs", err)
 		}
-		out.Step(style.Documentation, "Docs have been saved at - {{.path}}", out.V{"path": path})
+		out.Styled(style.Documentation, "Docs have been saved at - {{.path}}", out.V{"path": path})
```
Comment (priyawadhwa) [inline cmd/minikube/cmd/generate-docs.go:50]: out.step

File: pkg/minikube/bootstrapper/kubeadm/kubeadm.go [hunk 8]
```diff
@@ -319,7 +319,7 @@ func outputKubeadmInitSteps(logs io.Reader) {
 }
 
 // applyCNI applies CNI to a cluster. Needs to be done every time a VM is powered up.
-func (k *Bootstrapper) applyCNI(cfg config.ClusterConfig) error {
+func (k *Bootstrapper) applyCNI(cfg config.ClusterConfig, isInit bool) error {
```
Comment (medyagh) [inline pkg/minikube/bootstrapper/kubeadm/kubeadm.go:322]: how about rename it to `registerStep bool`

File: pkg/minikube/bootstrapper/kubeadm/kubeadm.go [hunk 9]
```diff
@@ -329,8 +329,13 @@ func (k *Bootstrapper) applyCNI(cfg config.ClusterConfig) error {
 		return nil
 	}
 
-	register.Reg.SetStep(register.ConfiguringCNI)
-	out.Step(style.CNI, "Configuring {{.name}} (Container Networking Interface) ...", out.V{"name": cnm.String()})
+	// when not on init, can run in parallel and break step output order
```
Comment (medyagh) [inline pkg/minikube/bootstrapper/kubeadm/kubeadm.go:332]: thanks for the comment

File: pkg/minikube/bootstrapper/kubeadm/kubeadm.go (hunk 8 above)
Comment (medyagh) [reply pkg/minikube/bootstrapper/kubeadm/kubeadm.go:322]: make this argument optional
so other callers can just called it

applyCNI
and if not True then it wont register step

**Review by medyagh — CHANGES_REQUESTED** [review]

**Review by medyagh — APPROVED** [review]

**Pull request merged**